crates/language/src/language.rs

```diff
@@ -890,20 +892,28 @@ impl LanguageRegistry {
         })
     }
 
+    /// context_info is a (worktree ID, &AppContext),
```

Comment:
This is not really a rustdoc for the method, it's a rustdoc for `(usize, &AppContext)` which has to be a struct with named fields, so let's create one — defining it will replace 80% of this comment already.

I know that there's already a `Settings::get` that suffers from it, but not sure we need to spread this across the code base.

Alternatively, we could remove a different oddity — `cx` buried in a tuple inside of an `Option` and pass it as another argument, directly — sure we would not need it in all cases, but at least this will look like the rest of the code base and no extra struct is needed, and `context_info` could be just a parameter named `workspace_id`?
That would be even better, since `context_info` looks like a very odd generic name for something that influences the language retrieval.

Reply:
And as mentioned in the [redundant comment of mine](https://github.com/zed-industries/zed/pull/7344#discussion_r1477144301), we seem not to want any `Option` at all here, since we have to consider the language overrides every time we try to determine the language.